ccc/github.py

```diff
@@ -63,6 +66,46 @@ def github_api_ctor(
         raise ValueError('failed to parse url: ' + str(github_url))
 
     session = github3.session.GitHubSession()
+    original_request = session.request
+
+    short = False
+
+    def intercepted_request(
+        method: str,
+        url: str,
+        **kwargs,
+    ):
+        nonlocal short
+
+        req = original_request(method, url, **kwargs)
+        if short:
+            return req
+
+        try:
+            es_client.store_document(
+                index='github_request',
+                body={
+                    'method': method,
+                    'url': url,
+                    'data': kwargs,
+                    'creation_date': datetime.datetime.now().isoformat(),
+                },
+            )
+        except:
+            logger.warning('unable to log github api request to Elasticsearch')
+            short = True
```

Comment:
We won't log for retries if the original request failed to log, right? Not an issue, I think, but is this really necessary?

Reply:
> We won't log for retries if the original request failed to log, right?

Correct.

However, it is not about single requests with retries but multiple requests.
In case logging fails (elasticsearch down, ...) we would write to stderr for every request, resulting in spam.
Therefore, shortcut logging for github-api object on first logging failure.